openmc/mgxs_library.py

```diff
@@ -0,0 +1,2577 @@
+
+        # Get the accepted shapes for this xs
+        shapes = [self.xs_shapes["[G]"], self.xs_shapes["[G][G']"]]
+
+        # Convert to a numpy array so we can easily get the shape for checking
+        prompt_nu_fission = np.asarray(prompt_nu_fission)
+        check_value('prompt_nu_fission shape', prompt_nu_fission.shape, shapes)
+        check_iterable_type('prompt_nu_fission', prompt_nu_fission, Real,
+                            max_depth=len(prompt_nu_fission.shape))
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
+
+        i = np.where(self.temperatures == temperature)[0][0]
+        self._prompt_nu_fission[i] = prompt_nu_fission
+        if np.sum(prompt_nu_fission) > 0.0:
+            self._fissionable = True
+
+    def set_delayed_nu_fission(self, delayed_nu_fission, temperature=294.):
+        """This method sets the cross section for this XSdata object at the
+        provided temperature.
+
+        Parameters
+        ----------
+        delayed_nu_fission: np.ndarray
+            Delayed-nu-fission Cross Section
+        temperature : float
+            Temperature (in units of Kelvin) of the provided dataset. Defaults
+            to 294K
+
+        See also
+        --------
+        openmc.mgxs_library.set_delayed_nu_fission_mgxs()
+
+        """
+
+        # Get the accepted shapes for this xs
+        shapes = [self.xs_shapes["[DG][G]"], self.xs_shapes["[DG][G][G']"]]
+
+        # Convert to a numpy array so we can easily get the shape for checking
+        delayed_nu_fission = np.asarray(delayed_nu_fission)
+        check_value('delayed_nu_fission shape', delayed_nu_fission.shape,
+                    shapes)
+        check_iterable_type('delayed_nu_fission', delayed_nu_fission, Real,
+                            max_depth=len(delayed_nu_fission.shape))
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
+
+        i = np.where(self.temperatures == temperature)[0][0]
+        self._delayed_nu_fission[i] = delayed_nu_fission
+        if np.sum(delayed_nu_fission) > 0.0:
+            self._fissionable = True
+
+    def set_inverse_velocity(self, inv_vel, temperature=294.):
+        """This method sets the inverse velocity for this XSdata object at the
+        provided temperature.
+
+        Parameters
+        ----------
+        inv_vel: np.ndarray
+            Inverse velocity in units of sec/cm.
+        temperature : float
+            Temperature (in Kelvin) of the data. Defaults to room temperature
+            (294K).
+
+        """
+
+        # Get the accepted shapes for this xs
+        shapes = [self.xs_shapes["[G]"]]
+
+        # Convert to a numpy array so we can easily get the shape for checking
+        inv_vel = np.asarray(inv_vel)
+        check_value('inverse_velocity shape', inv_vel.shape, shapes)
+        check_type('temperature', temperature, Real)
```

Comment:
These temperature checks (~30 overall) should be moved to a standalone method.

```
def check_temperature(temperature):
        check_type('temperature', temperature, Real)
        check_value('temperature', temperature, self.temperatures)
```

Then when calling it from code:
```
self.check_temperature(temperature)
```